verification.py

```diff
@@ -4623,8 +4658,31 @@ def run(self, output_dir, names=None, group=None):
             tests = [test for test in self.tests if test.group == group]
         else:
             tests = self.tests
-        for test in tests:
-            test.run(output_dir)
+
+        with concurrent.futures.ProcessPoolExecutor(max_workers=threads) as executor:
+            pbar = tqdm.tqdm(total=len(tests), desc="Queueing tests", leave=True)
+            futures = []
+            for test in tests:
```

Comment:
This doesn't work because you are not registering the futures, you're just running the test in the current thread. This is how to do it:
```
futures = [executor.submit(test.run, output_dir) for test in tests] 
pbar = tqdm.tqdm(total=len(futures), desc="Running tests", leave=True)
for future in concurrent.futures.as_completed(futures):
    pbar.update(1)
```

Reply:
I'm curious for knowledge's sake: how does it make a difference? I thought list comprehension worked in the same way as loops under the hood.

Reply:
It's not the list comprehension, it's the argument to submit: you had ``executor.submit(test.run(output_dir))`` whereas it has to be ``executor.submit(test.run, output_dir)``. So, you were executing the run method directly in the current thread rather than submitting a callable with arguments for future execution.